### ISSUE TITLE
Cannot install on Windows and Python39

### ISSUE THREAD

**Issue by tankorsmash** [swistakm/pyimgui#188]
Towards the bottom of the output (attached at the end of this issue) is a lot of repeats about `tp_print` not existing.

    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(77041): error C2039: 'tp_print': is not a member of '_typeobject'

It looks like `tp_print` was removed in Python 3.9 https://bugs.python.org/issue39361, and it sorta looks like any use of it in [python 3.x has been ignored](https://github.com/python/cpython/commit/2d5097663d7f80921fb07cdcd26c9d59cf71f1a2#diff-ec874d66bbcf13046b9928aa3716d14a93632f568b914eba32baf46ff4c9eb36R326),  so maybe just commenting out the lines would be enough to get it to compile.

This is the full output the `pip39 install imgui` command:

```
Defaulting to user installation because normal site-packages is not writeable
Collecting imgui
  Using cached imgui-1.2.0.tar.gz (830 kB)
Using legacy 'setup.py install' for imgui, since package 'wheel' is not installed.
Installing collected packages: imgui
    Running setup.py install for imgui ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Program Files\Python39\python39.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\asd\\AppData\\Local\\Temp\\pip-install-lvvnnu13\\imgui\\setup.py'"'"'; __file__='"'"'C:\\Users\\asd\\AppData\\Local\\Temp\\pip-install-lvvnnu13\\imgui\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\asd\AppData\Local\Temp\pip-record-p2njwoy2\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\asd\AppData\Roaming\Python\Python39\Include\imgui'
         cwd: C:\Users\asd\AppData\Local\Temp\pip-install-lvvnnu13\imgui\
    Complete output (104 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.9
    creating build\lib.win-amd64-3.9\imgui
    copying imgui\extra.py -> build\lib.win-amd64-3.9\imgui
    copying imgui\_compat.py -> build\lib.win-amd64-3.9\imgui
    copying imgui\__init__.py -> build\lib.win-amd64-3.9\imgui
    creating build\lib.win-amd64-3.9\imgui\integrations
    copying imgui\integrations\base.py -> build\lib.win-amd64-3.9\imgui\integrations
    copying imgui\integrations\cocos2d.py -> build\lib.win-amd64-3.9\imgui\integrations
    copying imgui\integrations\glfw.py -> build\lib.win-amd64-3.9\imgui\integrations
    copying imgui\integrations\opengl.py -> build\lib.win-amd64-3.9\imgui\integrations
    copying imgui\integrations\pygame.py -> build\lib.win-amd64-3.9\imgui\integrations
    copying imgui\integrations\pyglet.py -> build\lib.win-amd64-3.9\imgui\integrations
    copying imgui\integrations\sdl2.py -> build\lib.win-amd64-3.9\imgui\integrations
    copying imgui\integrations\__init__.py -> build\lib.win-amd64-3.9\imgui\integrations
    running egg_info
    writing imgui.egg-info\PKG-INFO
    writing dependency_links to imgui.egg-info\dependency_links.txt
    writing requirements to imgui.egg-info\requires.txt
    writing top-level names to imgui.egg-info\top_level.txt
    reading manifest file 'imgui.egg-info\SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    no previously-included directories found matching 'imgui-cpp\examples'
    no previously-included directories found matching 'imgui-cpp\extra_fonts'
    writing manifest file 'imgui.egg-info\SOURCES.txt'
    copying imgui\cimgui.pxd -> build\lib.win-amd64-3.9\imgui
    copying imgui\core.cpp -> build\lib.win-amd64-3.9\imgui
    copying imgui\core.h -> build\lib.win-amd64-3.9\imgui
    copying imgui\core.pxd -> build\lib.win-amd64-3.9\imgui
    copying imgui\core.pyx -> build\lib.win-amd64-3.9\imgui
    copying imgui\enums.pxd -> build\lib.win-amd64-3.9\imgui
    running build_ext
    building 'imgui.core' extension
    creating build\temp.win-amd64-3.9
    creating build\temp.win-amd64-3.9\Release
    creating build\temp.win-amd64-3.9\Release\config-cpp
    creating build\temp.win-amd64-3.9\Release\imgui-cpp
    creating build\temp.win-amd64-3.9\Release\imgui
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPYIMGUI_CUSTOM_EXCEPTION -Iimgui -Iconfig-cpp -Iimgui-cpp -IC:\Program Files\Python39\include -IC:\Program Files\Python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt /EHsc /Tpconfig-cpp/py_imconfig.cpp /Fobuild\temp.win-amd64-3.9\Release\config-cpp/py_imconfig.obj /FIpy_imconfig.h
    py_imconfig.cpp
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPYIMGUI_CUSTOM_EXCEPTION -Iimgui -Iconfig-cpp -Iimgui-cpp -IC:\Program Files\Python39\include -IC:\Program Files\Python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt /EHsc /Tpimgui-cpp/imgui.cpp /Fobuild\temp.win-amd64-3.9\Release\imgui-cpp/imgui.obj /FIpy_imconfig.h
    imgui.cpp
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPYIMGUI_CUSTOM_EXCEPTION -Iimgui -Iconfig-cpp -Iimgui-cpp -IC:\Program Files\Python39\include -IC:\Program Files\Python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt /EHsc /Tpimgui-cpp/imgui_demo.cpp /Fobuild\temp.win-amd64-3.9\Release\imgui-cpp/imgui_demo.obj /FIpy_imconfig.h
    imgui_demo.cpp
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPYIMGUI_CUSTOM_EXCEPTION -Iimgui -Iconfig-cpp -Iimgui-cpp -IC:\Program Files\Python39\include -IC:\Program Files\Python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt /EHsc /Tpimgui-cpp/imgui_draw.cpp /Fobuild\temp.win-amd64-3.9\Release\imgui-cpp/imgui_draw.obj /FIpy_imconfig.h
    imgui_draw.cpp
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPYIMGUI_CUSTOM_EXCEPTION -Iimgui -Iconfig-cpp -Iimgui-cpp -IC:\Program Files\Python39\include -IC:\Program Files\Python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt /EHsc /Tpimgui-cpp/imgui_widgets.cpp /Fobuild\temp.win-amd64-3.9\Release\imgui-cpp/imgui_widgets.obj /FIpy_imconfig.h
    imgui_widgets.cpp
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPYIMGUI_CUSTOM_EXCEPTION -Iimgui -Iconfig-cpp -Iimgui-cpp -IC:\Program Files\Python39\include -IC:\Program Files\Python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt -IE:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt /EHsc /Tpimgui/core.cpp /Fobuild\temp.win-amd64-3.9\Release\imgui/core.obj /FIpy_imconfig.h
    core.cpp
    imgui/core.cpp(30405): warning C4244: 'argument': conversion from 'Py_ssize_t' to 'int', possible loss of data
    imgui/core.cpp(44479): warning C4244: '=': conversion from 'Py_ssize_t' to 'int', possible loss of data
    imgui/core.cpp(44861): warning C4244: '=': conversion from 'Py_ssize_t' to 'int', possible loss of data
    imgui/core.cpp(48662): warning C4244: 'argument': conversion from 'double' to 'float', possible loss of data
    imgui/core.cpp(76878): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76889): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76900): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76911): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76922): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76935): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76946): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76957): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76968): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76980): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76989): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(76995): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(77001): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(77007): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(77013): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(77021): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(77026): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(77041): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(77054): error C2039: 'tp_print': is not a member of '_typeobject'
    c:\program files\python39\include\cpython/object.h(193): note: see declaration of '_typeobject'
    imgui/core.cpp(83379): warning C4996: 'PyUnicode_FromUnicode': deprecated in 3.3
    c:\program files\python39\include\cpython/unicodeobject.h(551): note: see declaration of 'PyUnicode_FromUnicode'
    imgui/core.cpp(83994): warning C4996: '_PyUnicode_get_wstr_length': deprecated in 3.3
    c:\program files\python39\include\cpython/unicodeobject.h(446): note: see declaration of '_PyUnicode_get_wstr_length'
    imgui/core.cpp(84010): warning C4996: '_PyUnicode_get_wstr_length': deprecated in 3.3
    c:\program files\python39\include\cpython/unicodeobject.h(446): note: see declaration of '_PyUnicode_get_wstr_length'
    imgui/core.cpp(84267): warning C4996: 'PyUnicode_FromUnicode': deprecated in 3.3
    c:\program files\python39\include\cpython/unicodeobject.h(551): note: see declaration of 'PyUnicode_FromUnicode'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Program Files\Python39\python39.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\asd\\AppData\\Local\\Temp\\pip-install-lvvnnu13\\imgui\\setup.py'"'"'; __file__='"'"'C:\\Users\\asd\\AppData\\Local\\Temp\\pip-install-lvvnnu13\\imgui\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\asd\AppData\Local\Temp\pip-record-p2njwoy2\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\asd\AppData\Roaming\Python\Python39\Include\imgui' Check the logs for full command output.
WARNING: You are using pip version 20.2.3; however, version 20.2.4 is available.
You should consider upgrading via the 'C:\Program Files\Python39\python39.exe -m pip install --upgrade pip' command.

```

**Comment by swistakm** [comment]
Pipeline has been finally fixed, so once all build artifacts arrive I will push new release to PyPI and close this issue.

**Comment by swistakm** [comment]
Released as `1.3.0`:
- [release notes](https://github.com/swistakm/pyimgui/releases/tag/1.3.0)
- [available distributions](https://pypi.org/project/imgui/1.3.0/#files)
